server/middleware/authentication/verifyUser.js

```diff
@@ -0,0 +1,20 @@
+const { verify } = require('../../utils/jwt');
+const boomify = require('../../utils/boomify');
+
+const verifyUser = async (req, res, next) => {
+  const { token } = req.cookies;
+  if (token) {
+    try {
+      const { decoded } = await verify(token);
+      req.userId = decoded;
+      next();
+    } catch (error) {
+      res.clearCookie('token');
+      next(boomify(401, 'Unauthorized Error', 'unauthorized User'));
```

Comment:
I don't think there's any need to clear the cookies when there's an error in the catch, and instead of throw a `401 Unauthorized Error` you should `next(error)` because most likely this will be an error with the server `500 Internal Server Error` not `401`
```suggestion
      next(error);
```

Reply:
What if there was a manipulation in the token, we will not be able to catch it 
also, if the token is Manipulated on it, It should be deleted